Extra indenting in a nested ternary with offsetTernaryExpressions: true from Standard JS config

<!--
    ESLint adheres to the Open JS Foundation Code of Conduct:
    https://eslint.org/conduct

    This template is for bug reports. If you are here for another reason, please see below:

    1. To propose a new rule: https://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a rule change: https://eslint.org/docs/developer-guide/contributing/rule-changes
    3. To request a change that is not a bug fix, rule change, or new rule: https://eslint.org/docs/developer-guide/contributing/changes
    4. If you have any questions, please stop by our chatroom: https://eslint.org/chat

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->


**Tell us about your environment**

<!--
    If you are using ESLint v6.5.0 or later, you can run ESLint with the `--env-info` flag and paste the output here.
-->

* **ESLint Version:** 7.16.0
* **Node Version:** 14.5.3
* **npm Version:** 6.14.9

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

Here is the configuration from my work in <https://github.com/brodybits/react-native-module-init/pull/85> but with Prettier items removed since they are not needed to reproduce this issue:

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
parserOptions:
  ecmaVersion: 9

extends:
  - standard

plugins:
  - jest

env: { jest/globals: true }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Reduced reproduction code that I made in a local `mock.test.js` file:

<!-- Paste the source code below: -->
```js
const mockCallSnapshot = []

jest.mock('execa', () => (cmd, args, opts) => {
  mockCallSnapshot.push({ execa: [cmd, args, opts] })
  return cmd === 'git'
    ? args[1] === 'user.email'
      ? Promise.resolve({ stdout: 'alice@example.com' })
      : Promise.resolve({ stdout: 'Alice' })
    : Promise.resolve()
})
```

<!-- Paste the command you used to run ESLint: -->
```bash
npx eslint mock.test.js
```

**What did you expect to happen?**

no errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/brodybits/dev/react-native-module-init/mock.test.js
  7:1  error  Expected indentation of 8 spaces but found 6  indent
  8:1  error  Expected indentation of 8 spaces but found 6  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

`npx eslint --fix mock.test.js` updates the code with extra indenting in the nested ternary like this:

```js
const mockCallSnapshot = []

jest.mock('execa', () => (cmd, args, opts) => {
  mockCallSnapshot.push({ execa: [cmd, args, opts] })
  return cmd === 'git'
    ? args[1] === 'user.email'
        ? Promise.resolve({ stdout: 'alice@example.com' })
        : Promise.resolve({ stdout: 'Alice' })
    : Promise.resolve()
})
```

I was able to reproduce this issue more simply with the following fragment in the test suite:

```js
        {
            code: unIndent`
              condition1
                ? condition2
                  ? Promise.resolve(1)
                  : Promise.resolve(2)
                : Promise.resolve(3)
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
```

**Are you willing to submit a pull request to fix this bug?**

Yes.

I was able to make a one-line fix based on bb6cf50826 (PR #12556) by @sheerun:

<details>
<summary>One-line fix</summary>

```diff
--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1178,6 +1178,7 @@ module.exports = {
                     offsets.setDesiredOffset(colonToken, firstToken, 1);
 
                     offsets.setDesiredOffset(firstConsequentToken, firstToken,
+                        firstConsequentToken.type === "Punctuator" &&
                         options.offsetTernaryExpressions ? 2 : 1);
 
                     /*
```

</details>

This issue is in follow-up to the offsetTernaryExpressions option contributed in PR #12556 and has led to <https://github.com/standard/standard/issues/1624>.

/cc @sheerun

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1178,6 +1178,7 @@ module.exports = {
                     offsets.setDesiredOffset(colonToken, firstToken, 1);
 
                     offsets.setDesiredOffset(firstConsequentToken, firstToken,
+                        firstConsequentToken.type === "Punctuator" &&
                         options.offsetTernaryExpressions ? 2 : 1);
 
                     /*

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -2216,6 +2216,16 @@ ruleTester.run("indent", rule, {
         },
         {
             code: unIndent`
+              condition1
+                ? Promise.resolve(1)
+                : condition2
+                  ? Promise.resolve(2)
+                  : Promise.resolve(3)
+            `,
+            options: [2, { offsetTernaryExpressions: true }]
+        },
+        {
+            code: unIndent`
               condition
               \t? () => {
               \t\t\treturn true

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -2206,6 +2206,16 @@ ruleTester.run("indent", rule, {
         },
         {
             code: unIndent`
+              condition1
+                ? condition2
+                  ? Promise.resolve(1)
+                  : Promise.resolve(2)
+                : Promise.resolve(3)
+            `,
+            options: [2, { offsetTernaryExpressions: true }]
+        },
+        {
+            code: unIndent`
               condition
               \t? () => {
               \t\t\treturn true

